sql/core/src/main/java/org/apache/spark/sql/sources/v2/DataSourceV2.java

```diff
@@ -23,8 +23,9 @@
  * The base interface for data source v2. Implementations must have a public, 0-arg constructor.
  *
  * Note that this is an empty interface. Data source implementations should mix-in at least one of
- * the plug-in interfaces like {@link ReadSupport} and {@link WriteSupport}. Otherwise it's just
- * a dummy data source which is un-readable/writable.
+ * the provider interfaces like {@link BatchReadSupportProvider} or
```

Comment:
I think this should say that implementations can mix-in BatchReadSupportProvider etc. to provide anonymous tables for use with the DF writer API. When `TableCatalog` is in, this won't be required at all.